aospy/test/test_utils_times.py

```diff
@@ -421,6 +315,94 @@ def test_assert_has_data_for_time():
         _assert_has_data_for_time(da, start_date_bad, end_date_bad)
 
 
+CFTIME_DATE_TYPES = {
```

Comment:
Ah I see here you test all of the calendar types (c.f. my other comment).   If you think that it's sufficient to do so at this lower level, then fine by me.

Reply:
I think you're right to be strive to be thorough in testing all kinds of `cftime.datetime` objects at this low level.  I could be more thorough in these tests (particularly for `infer_year` and `maybe_convert_to_index_date_type`).  I'll improve those.